app/grandchallenge/algorithms/views.py

```diff
@@ -323,10 +316,29 @@ def get_context_data(self, *args, **kwargs):
 
     def get_success_url(self):
         return reverse(
-            "algorithms:jobs-list", kwargs={"slug": self.kwargs["slug"]}
+            "algorithms:execution-session-detail",
+            kwargs={"slug": self.kwargs["slug"], "pk": self.object.pk},
         )
 
 
+class AlgorithmExecutionSessionDetail(
+    LoginRequiredMixin, ObjectPermissionRequiredMixin, DetailView
+):
+    model = RawImageUploadSession
+    template_name = "algorithms/executionsession_detail.html"
+    permission_required = "cases.view_rawimageuploadsession"
+    raise_exception = True
+
+    @property
+    def algorithm(self) -> Algorithm:
+        return Algorithm.objects.get(slug=self.kwargs["slug"])
```

Comment:
maybe use `get_object_or_404` here, just in case the algorithm does not exist.

Reply:
Ah, of course! We use this in a lot of places... I will make the change.

Reply:
Fixed in 97e8c65